### PR TITLE
Manage yumrepos via data

### PR DESCRIPTION
**Pull request by lamawithonel**
This commit adds parameters to manage Yumrepo resources via data.  The
primary parameter, `repos`, is simply a hash of Yumrepo resources for
digestion by an each loop similar to an old `create_resources()` call.
It does nothing on its own, acting only as a source of data.  The other
parameters control the behavior of the resource instantiation loop.

The parameters `managed_repos`, `os_default_repos`, and
`repo_exclusions` each take an array of key names from the `repos` hash.
They are then combined to select a set of repos from the `repos` hash
for instantiation.  The formula takes one of the following two forms:

    `managed_repos` - `repo_exclusions`

OR, if `manage_os_default_repos` is set to `true`

    `managed_repos` + `os_default_repos` - `repo_exclusions`

Data is included for most of the CentOS default repositories, with the
exception of the "archive" repos.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@
   default_facts.merge!(YAML.load(File.read(File.expand_path('../default_facts.yml', __FILE__)))) if File.exist?(File.expand_path('../default_facts.yml', __FILE__))
   default_facts.merge!(YAML.load(File.read(File.expand_path('../default_module_facts.yml', __FILE__)))) if File.exist?(File.expand_path('../default_module_facts.yml', __FILE__))
   c.default_facts = default_facts
+  c.hiera_config = File.expand_path(File.join(__FILE__, '../fixtures/hiera.yaml'))
```
Comment (vinzent) [inline spec/spec_helper.rb]: i think this will be overriden by modulesync.

Comment (lamawithonel) [inline spec/spec_helper.rb]: As far as `spec_helper,rb` goes, I've filed a modulesync_config PR (voxpupuli/modulesync_config#317).

**Comment by lamawithonel** [comment]
@vinzent This latest version adds a parameter to toggle management of the OS repos, defaulting to `false`.  The spec test for the default case now checks that it includes *zero* Yumrepo resources.

File: manifests/init.pp [hunk 2]
```diff
@@ -1,16 +1,108 @@
...
+#   this list to select `Yumrepo` resources from the `repos` hash for instantiation.  Defaults are
+#   set in the module's Hiera data.
+#
+#   @note This only indicates the *managed* state of the repos, the `ensure` state must be managed
+#     in the `repos` data.
+#
+# @param manage_os_default_repos [Boolean]
+#   Whether or not to add an operating system's default repos to the `managed_repos` array.
+#
+#   @note Currently only valid on CentOS.
```
Comment (hunner) [inline manifests/init.pp]: This parameter is valid on other OSs, but `os_default_repos` will provide no defaults for any OS other than CentOS 6/7. Perhaps say something like "currently only provides defaults for CentOS" on the `os_default_repos` parameter instead?

**Comment by lamawithonel** [comment]
I've added a whole bunch of examples to the README.  Hopefully they shed some light on why this is useful.

**Pull request merged**